provider/boltmem/boltmem.go

```diff
@@ -0,0 +1,426 @@
+package boltmem
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/boltdb/bolt"
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	bktNotifies = []byte("notifies")
```

Comment:
Is there a chance we can name this "notifications", since "notifies" confuses me every time I read it? "notifies" is just a conjugation of the verb "to notify", but never a valid noun.

Just saying this now since this will be persisted and will thus be harder to change later on.